qiskit/providers/ibmq/credentials/__init__.py

```diff
@@ -81,12 +81,12 @@ def discover_credentials(
         ('environment variables', (read_credentials_from_environ, {})),
         ('qiskitrc', (read_credentials_from_qiskitrc,
                       {'filename': qiskitrc_filename}))
-    ])
+    ])  # type: MutableMapping[str, Any]
```

Comment:
Why is this a `MutableMapping` and not `Dict`?

Reply:
At first, I was trying to be more specific with the return type and `mypy` would complain about `Dict`, because it expected tuples as the entries. `MutableMapping` was a suggested solution. After changing the type comment to something simpler (the actual type would be way too long and it is also mentioned in line 77), I left it as `MutableMapping`. It looks like using `Dict` is no issue if the type is simple.

Overlooked it, but we could just use `OrderedDict` as the type, since it is imported...

Reply:
In that case line 75 should also be `OrderedDict`?